lib/src/db.rs

```diff
@@ -107,7 +112,7 @@ impl Db {
     #[instrument(skip(self))]
     fn set_propvals(&self, subject: &str, propvals: &PropVals) -> AtomicResult<()> {
         let resource_bin = bincode::serialize(propvals)?;
-        let subject_bin = bincode::serialize(subject)?;
+        let subject_bin = subject.as_bytes();
```

Comment:
This may cause DB corruption, as I'm not sure that these serialize in the same way.

If it does corrup between versions, we need some form of (automated) migration.

Reply:
Yes, it's breaking change and will corrupt the database:
```rust 
let subject="https://atomicdata.dev/datatypes/float";
subject.as_bytes() // new way
[104, 116, 116, 112, 115, 58, 47, 47, 97, 116, 111, 109, 105, 99, 100, 97, 116, 97, 46, 100, 101, 118, 47, 100, 97, 116, 97, 116, 121, 112, 101, 115, 47, 102, 108, 111, 97, 116]
bincode::serialize(subject).unwrap() //old way
[38, 0, 0, 0, 0, 0, 0, 0, 104, 116, 116, 112, 115, 58, 47, 47, 97, 116, 111, 109, 105, 99, 100, 97, 116, 97, 46, 100, 101, 118, 47, 100, 97, 116, 97, 116, 121, 112, 101, 115, 47, 102, 108, 111, 97, 116]
bincode::serialize("https://atomicdata.dev/").unwrap() //mind prefix 
[23, 0, 0, 0, 0, 0, 0, 0, 104, 116, 116, 112, 115, 58, 47, 47, 97, 116, 111, 109, 105, 99, 100, 97, 116, 97, 46, 100, 101, 118, 47]
```
In the old way, serialisation with the same prefix URL will serialize into different "buckets/leafs" - this prevents the use of scan_prefix use or leveraging internal structure for sled::Tree. 
So, what is the approach to data migrations?

Reply:
> In the old way, serialisation with the same prefix URL will serialize into different "buckets/leafs" - this prevents the use of scan_prefix use or leveraging internal structure for sled::Tree.

Can't say I fully understand this. `scan_prefix` should also work if there is a shared prefix, in the URL, right? I think I'm missing something here.

> So, what is the approach to data migrations?

I think, sooner or later, we'll need to store the version in a `Tree`, and check this version on booting. If there is a mismatch, the relevant migration script(s) will run. 

So in this case, we iterate over all resources, use `bincode::deserialize` and then put them back with `.as_bytes()`.

We should add a new file in the `db` folder named `migrations.rs`. You then create a function `v0.31_to_v0.32`, which takes the `Db` as its only argument.

When running `Db::init`, we should open a new `Tree` called `Config`, which has a key called `version`, which is checked. If there is no version, we call the new migration.

What do you think?